Offices/Offices.Application/Services/DoctorService.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using AutoMapper;
+using Offices.Application.Models;
+using Offices.Application.Services.Abstractions;
+using Offices.Data.Entities;
+using Offices.Data.Exceptions;
+using Offices.Data.Repositories.Abstractions;
+
+namespace Offices.Application.Services
+{
+    public class DoctorService: IDoctorService
+    {
+        private readonly IGenericRepository<Doctor> _doctorRepository;
+        private readonly IGenericRepository<Office> _officeRepository;
+        private readonly IMapper _mapper;
+
+        public DoctorService(IGenericRepository<Doctor> doctorRepository, IGenericRepository<Office> officeRepository, IMapper mapper)
+        {
+            _officeRepository = officeRepository;
+            _doctorRepository = doctorRepository;
+            _mapper = mapper;
+        }
+
+        public async Task Create(CreateDoctorModel createDoctorModel)
+        {
+            // If specified office is not active, can't create entity
+            if ((await _officeRepository.GetAsync(createDoctorModel.OfficeId)) == null)
```

Comment:
either move this call to the separate line or move it to a private method

Reply:
Don't put any invocations to the if statement, it is unreadable

Reply:
Fixed